File: dbflow/src/main/java/com/raizlabs/android/dbflow/runtime/ModelNotifier.java
package com.raizlabs.android.dbflow.runtime;

import androidx.annotation.NonNull;

import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.structure.ModelAdapter;

/**
 * Interface for defining how we notify model changes.
 */
public interface ModelNotifier {

    <T> void notifyModelChanged(@NonNull T model, @NonNull ModelAdapter<T> adapter,
                                @NonNull BaseModel.Action action);

    <T> void notifyTableChanged(@NonNull Class<T> table, @NonNull BaseModel.Action action);

    TableNotifierRegister newRegister();
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/runtime/ModelNotifier.kt
package com.raizlabs.android.dbflow.runtime

import com.raizlabs.android.dbflow.structure.ChangeAction
import com.raizlabs.android.dbflow.structure.ModelAdapter

/**
 * Interface for defining how we notify model changes.
 */
interface ModelNotifier {

    fun <T : Any> notifyModelChanged(model: T, adapter: ModelAdapter<T>, action: ChangeAction)

    fun <T : Any> notifyTableChanged(table: Class<T>, action: ChangeAction)

    fun newRegister(): TableNotifierRegister
}
